lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -25,13 +25,26 @@
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 3
+LIBPATCH = 4
 
 
 logger = logging.getLogger(__name__)
 RAFT_STATE_ENDPOINT = "v1/sys/storage/raft/autopilot/state"
 
 
+class VaultClientError(Exception):
+    """Error raised when interacting with Vault."""
```

Comment:
[hvac.exceptions.VaultError](https://hvac.readthedocs.io/en/stable/_modules/hvac/exceptions.html) is exactly this class, so there is no need for this.

Reply:
Correct but it's owned by the hvac lib, it doesn't make sense to for the charm code to have to know of exceptions owned by its's libs' libs.

Reply:
That makes sense. It's also possible to import the same VaultError from the client instead of hvac, but that's more of a javascript thing that doesn't happen in python.

Reply:
Done, we now return None in the `sign_...` method and there's no need for this error.